lua/neo-tree/ui/renderer.lua

```diff
@@ -17,6 +17,27 @@ local default_popup_size = { width = 60, height = "80%" }
 local floating_windows = {}
 local draw, create_window, create_tree, render_tree
 
+-- clean up neotree buffers created by :mksession
+local cleaned_up = false
+local clean_neotree_buffers = function()
+  if cleaned_up then
+    return
+  end
+  local clean_buffer_with_prefix = function(prefix)
+    for bufnr = 1, vim.fn.bufnr("$") do
+      local bufname = vim.fn.bufname(bufnr)
+      if string.sub(bufname, 1, #prefix) == prefix then
```

Comment:
This could be more precise to avoid mistakes. This code is a good way to test if it is a neo-tree buffer:
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/245cf1e68840defcc75a16297740f6203f5a045d/lua/neo-tree/setup/init.lua#L497-L498

Reply:
I saw that the buffer created by neo-tree is named `neo-tree filesystem [1]`, but this does not seem to match the pattern in the init file (which requires 4 digits inside the square bracket). Should I remove those `%d`?

Reply:
Oh yeah, this pattern was specifically for "position=current" windows which have a 4 digit id. We really want a pattern that has any number of digits in the second bracket, which might be this:
```lua
"neo%-tree [^ ]+ %[%d+%]"
```